src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -119,22 +119,24 @@ internal TypeWithAnnotations AsNotAnnotated()
             return Create(Type, NullableAnnotation.NotAnnotated, CustomModifiers);
         }
 
-        internal bool IsPossiblyNullableTypeTypeParameter()
-        {
-            return NullableAnnotation.IsNotAnnotated() &&
-                (Type?.IsPossiblyNullableReferenceTypeTypeParameter() == true ||
-                 Type?.IsNullableTypeOrTypeParameter() == true);
-        }
-
         internal NullableAnnotation GetValueNullableAnnotation()
         {
-            if (IsPossiblyNullableTypeTypeParameter())
+            if (NullableAnnotation.IsAnnotated())
+            {
+                return NullableAnnotation;
+            }
+
+            if (Type is null)
+            {
+                return NullableAnnotation.NotAnnotated;
+            }
+
+            if (Type.IsPossiblyNullableReferenceTypeTypeParameter())
```

Comment:
TypeType ... makes sense but still gets me at a glance. Would `IsDefaultTypeParameter` work here?

Reply:
This is an existing method so I'll leave the name as is to avoid extra confusion since it's hard enough to keep track of the various methods we have currently.

---
In reply to: [455912988](https://github.com/dotnet/roslyn/pull/45993#discussion_r455912988) [](ancestors = 455912988)